src/inmanta/data/paging.py

```diff
@@ -128,6 +143,27 @@ async def count_items_for_paging(
         return PagingCounts(total=result[0]["count_total"], before=result[0]["count_before"], after=result[0]["count_after"])
 
 
+class CompileReportPagingCountsProvider(PagingCountsProvider):
+    def __init__(self, data_class: Type[Compile]) -> None:
```

Comment:
Why is this a parameter at all? afaik there is only one `Type[Compile]`, so it can just as well be hardcoded. 


I think in fact, this could be

```python
class CompileReportPagingCountsProvider(PagingCountsProvider):

    async def count_items_for_paging(
        self,
        environment: uuid.UUID,
        database_order: DatabaseOrder,
        first_id: Optional[Union[uuid.UUID, str]] = None,
        last_id: Optional[Union[uuid.UUID, str]] = None,
        start: Optional[Any] = None,
        end: Optional[Any] = None,
        **query: Any,
      ) -> PagingCounts:
          return await Compile.count_items_for_paging(...)
          
```

Or even have `Compile` extend `PagingCountsProvider` itself